ocaml/parsing/parser.mly

```diff
@@ -221,40 +232,41 @@ let wrap_exp_once exp loc =
 
 type modes = Local | Unique | Once
 
-let mkexp_with_mode ~loc ~kwd_loc flag exp =
-  match flag with
-  | Local -> mkexp_stack exp ~loc ~kwd_loc
-  | Unique -> mkexp_unique exp ~loc ~kwd_loc
-  | Once -> mkexp_once exp ~loc ~kwd_loc
+(* [loc] is the location of the whole expression including the extension node.
+    If not given, will use [exp.pexp_loc] (but ghost).
+    The extension node will always have non-ghost (but filled-in) location.
+ *)
+let mkexp_with_mode ?loc (mode, kwd_loc) exp =
+  match mode with
+  | Local -> mkexp_stack exp ?loc ~kwd_loc
+  | Unique -> mkexp_unique exp ?loc ~kwd_loc
+  | Once -> mkexp_once exp ?loc ~kwd_loc
 
-let mkexp_with_modes flags exp loc =
+let mkexp_with_modes modes exp =
```

Comment:
I wonder if we should name this `ghexp_with_modes` to more strongly suggest that the location minted is a ghost location. This matches the naming scheme used for a few other bindings in the file.